protocol/flows/src/flowcontext/orphans.rs

```diff
@@ -156,6 +157,36 @@ impl OrphanBlocksPool {
                 i += 1;
             }
         }
+
+        // Next, search for root blocks which are processable
+        let mut roots = Vec::new();
+        for block in self.orphans.values() {
+            let mut processable = true;
+            for p in block.block.header.direct_parents().iter().copied() {
+                if self.orphans.contains_key(&p) || consensus.async_get_block_status(p).await.is_none_or(|s| s.is_header_only()) {
+                    processable = false;
+                    break;
+                }
+            }
+            if processable {
+                roots.push(block.block.clone());
+            }
+        }
+
+        // Now process the roots and unorphan their descendents
+        let mut virtual_processing_tasks = Vec::with_capacity(roots.len());
+        let mut queued_hashes = Vec::with_capacity(roots.len());
+        for root in roots {
+            let root_hash = root.hash();
+            let BlockValidationFutures { block_task: _, virtual_state_task: root_task } = consensus.validate_and_insert_block(root);
+            virtual_processing_tasks.push(root_task);
+            queued_hashes.push(root_hash);
+            let (unorphan_blocks, _, unorphan_tasks) = self.unorphan_blocks(consensus, root_hash).await;
+            virtual_processing_tasks.extend(unorphan_tasks);
+            queued_hashes.extend(unorphan_blocks.into_iter().map(|b| b.hash()));
```

Comment:
Avoid names that are also functions. Also avoid single letter names.

```suggestion
            let (unorphaning_blocks, _, unorphan_tasks) = self.unorphan_blocks(consensus, root_hash).await;
            virtual_processing_tasks.extend(unorphan_tasks);
            queued_hashes.extend(unorphaning_blocks.into_iter().map(|block| block.hash()));
```